src/multibody/model.hpp

```diff
@@ -100,9 +100,13 @@ namespace se3
     std::vector<Frame> frames;
     
     /// \brief Vector of subtrees.
-    /// subtree[j] corresponds to the subtree supported by the joint j.
+    /// subtrees[j] corresponds to the subtree supported by the joint j.
```

Comment:
I do not agree with the choice of modifying (making more complex) model to implement a marginal algorithm. I dont think that it is a good strategy in principle. 
Here, the subtree structure should be given as an argument to the algorithm.